android/app/src/main/java/app/organicmaps/SplashActivity.java

```diff
@@ -157,6 +157,8 @@ public void processNavigation()
     // https://github.com/organicmaps/organicmaps/issues/6944
     final Intent intent = Objects.requireNonNull(getIntent());
     intent.setComponent(new ComponentName(this, DownloadResourcesLegacyActivity.class));
+    // Forward only FLAG_ACTIVITY_FORWARD_RESULT and FLAG_GRANT_READ_URI_PERMISSION flags.
+    intent.setFlags(intent.getFlags() & (Intent.FLAG_ACTIVITY_FORWARD_RESULT | Intent.FLAG_GRANT_READ_URI_PERMISSION));
```

Comment:
nit: Comment repeats what I already see in the code. Write that we have crashes otherwise, maybe add the link on previous refactoring PR.

Reply:
+1 to a more informative comment